test/compojure/api/resource_test.clj

```diff
@@ -134,6 +134,31 @@
       (handler {:request-method :get, :query-params {:x "10"}}) => (has-body {:total 10})
       (handler {:request-method :post, :query-params {:x "1"}}) => (has-body {:total 1}))))
 
+(let [req-error (atom nil)
+      res-error (atom nil)
+      result    (atom nil)
+      handler   (resource
+                 {:parameters {:query-params {:x Long}}
+                  :responses {200 {:schema {:total (s/constrained Long pos? 'pos)}}}
+                  :handler (fn [{{x :x} :query-params} respond _]
+                             (future
+                               (respond (ok {:total x}))))})]
+  @(handler  {:query-params {:x 1}}
+             #(reset! result %)
+             nil)
+  @(handler  {:query-params {:x -1}}
+             #(identity %)
+             #(reset! res-error %))
+  (handler  {:query-params {:x "x"}}
+            #(identity %)
+            #(reset! req-error %))
+
```

Comment:
This looks a bit ugly, but the asynchronous computation has to be forced somehow. We discussed this with @kalekale and couldn't think of any obviously better solution, so let's have it like this.